node/taskcommand.ts

```diff
@@ -33,12 +34,16 @@ export class TaskCommand {
                     var val = this.properties[key];
                     if (val) {
                         cmdStr += key + '=' + val + ';';
+                        if(key === 'secret' && val === 'true') {
+                            containsSecret = true;
+                        }
                     }                    
                 }
             }
         }
 
-        cmdStr += ']' + this.message;
+        cmdStr += ']';
+        cmdStr += containsSecret? '********' : this.message;
```

Comment:
SetVariable() calls 
command('task.setvariable', { 'variable': name || '', 'secret': (secret || false).toString() }, varValue) which write out the value to the console/log even if it is a secret. I fixed it in the toString() but we could do it in the command() function as well: https://github.com/Microsoft/vsts-task-lib/blob/master/node/task.ts#L618